### PR TITLE
Force xslt

### PR DESCRIPTION
**Pull request by marksreeves**
XSLT must take place in WComponents, not in a browser. This requires that we remove the ability to avoid the XSLT interceptor. This change removes the use of the PARAM without removing any of the public API, thereby allowing us to deprecate the public API for removal in v2.

Forms part of #639.

@jonathanaustin PTAL

### CODE REVIEW

File: wcomponents-core/src/test/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor_Test.java [hunk 1]
```diff
@@ -59,26 +59,12 @@ public static void tearDownClass() {
 	}
 
 	/**
```
Comment (jonathanaustin) [inline wcomponents-core/src/test/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor_Test.java]: You could keep this test, with setting the parameter to false and prove it still transforms.

File: wcomponents-lde/pom.xml [hunk 2]
```diff
@@ -43,8 +43,15 @@
 			<artifactId>junit</artifactId>
 		</dependency>
```
Comment (jonathanaustin) [inline wcomponents-lde/pom.xml]: wcomponents-core has a "dummy" xslt for testing the "transform". What do you think of using something similar for this test and then it wont be dependant on the "theme"...

File: wcomponents-core/src/test/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor_Test.java (hunk 1 above)
Comment (marksreeves) [inline wcomponents-core/src/test/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor_Test.java]: Not really happy with this. It is OK to prove the change has been made but not really useful after the initial pull.

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java [hunk 3]
```diff
@@ -323,6 +323,7 @@
 
 	/**
 	 * Whether to perform server-side XSLT.
```
Comment (marksreeves) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/ConfigurationProperties.java]: Yep, ok.

File: wcomponents-lde/pom.xml (hunk 2 above)
Comment (marksreeves) [inline wcomponents-lde/pom.xml]: This change was in line with wcomponents-examples. I am not really keen on having a dummy XSLT. We have a process which relies on an XSLT processor and we build XSLT so why not use the one we already have?

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor.java [hunk 4]
```diff
@@ -93,7 +93,7 @@
 	 */
 	@Override
 	public void preparePaint(final Request request) {
-		doTransform = isPerformTransform();
+		doTransform = true;
```
Comment (marksreeves) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/container/TransformXMLInterceptor.java]: Yeah, OK. It is unlikely that the header would change between requests within the same session.

**Comment by marksreeves** [comment]
Schema changes to fix #1036.

**Pull request merged**